google/cloud/functions/internal/parse_options.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/functions/internal/parse_options.h"
+#include <boost/program_options.hpp>
+#include <iostream>
+
+namespace google::cloud::functions_internal {
+inline namespace FUNCTIONS_FRAMEWORK_CPP_NS {
+namespace po = boost::program_options;
+
+auto constexpr kDefaultPort = 8080;
+
+po::variables_map ParseOptions(int& argc, char const* const argv[]) {
+  // Validate a port value
+  auto validate_port = [](int value) {
+    if (value < std::numeric_limits<std::uint16_t>::min() ||
+        value > std::numeric_limits<std::uint16_t>::max()) {
+      std::ostringstream os;
+      os << "The PORT environment variable value (" << value
+         << ") is out of range.";
+      throw std::invalid_argument(std::move(os).str());
+    }
+  };
+  // Initialize the default port with the value from the "PORT" environment
+  // variable or with 8080.
+  auto port = [&]() -> std::uint16_t {
+    auto const* env = std::getenv("PORT");
+    if (env == nullptr) return kDefaultPort;
+    auto value = std::stoi(env);
+    validate_port(value);
+    return static_cast<std::uint16_t>(value);
+  }();
```

Comment:
Perhaps this should just return an `int` and leave the validation for when we have the final value?

Reply:
Done.